poh-bundle-linux-x64/oh.sh

```diff
@@ -1,68 +1,259 @@
 #!/bin/bash
+#
+# Open Hospital (www.open-hospital.org)
+# Copyright © 2006-2020 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+#
+# Open Hospital is a free and open source software for healthcare data management.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# https://www.gnu.org/licenses/gpl-3.0-standalone.html
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+######## Environment check:
+
+######## OPENHOSPITAL Configuration
+# POH_PATH is the directory where Portable OpenHospital files are located
+# POH_PATH=/usr/local/PortableOpenHospital
+
+# name of this shell script
+SCRIPT_NAME="oh.sh"
+
+# set current dir
+CURRENT_DIR=$PWD
+
+if [ -z $POH_PATH ]; then
+	echo "Warning - POH_PATH not found. Please set it up properly."
+		if [ ! -f ./$SCRIPT_NAME ]; then
+		echo "Error - oh.sh not found in the current PATH. Please cd the directory where POH was unzipped or set up POH_PATH properly."
+		exit 0
+		fi
+	POH_PATH=$PWD
+fi
+
+######## Software configuration - change at your own risk :-)
 
 MYSQL_DIR="mysql-5.7.30-linux-glibc2.12-x86_64"
-JAVA_DIR="jdk-11.0.7+10-jre"
+MYSQL_URL="https://downloads.mysql.com/archives/get/p/23/file"
+
+JAVA_DISTRO="OpenJDK11U-jdk_x64_linux_openj9_11.0.8_10_openj9-0.21.0"
+JAVA_URL="https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.8%2B10_openj9-0.21.0"
+JAVA_DIR="jdk-11.0.8+10"
+
+MYSQL_PORT=3307
+MYSQL_SOCKET="var/run/mysqld/mysql.sock"
+DB_CREATE_SQL="database.sql"
+DB_ARCHIVED_SQL="database.sql.imported"
+
 OH_DIR="oh"
+DATABASE_NAME="oh"
+DATABASE_USER="isf"
+DATABASE_PASSWORD="isf123"
 DICOM_DEFAULT_SIZE="4M"
 
-where_i_am=$(pwd)
-cd $(dirname $0)
-POH_PATH=$(pwd)
-cd $where_i_am
+######################## DO NOT EDIT BELOW THIS LINE ########################
+
+cd $POH_PATH
+
+echo ""
+echo "***************************************************"
+echo "* Warning - Do not run this script as root user ! *"
+echo "***************************************************"
+echo ""
```

Comment:
```suggestion
if [ "$EUID" -e 0 ]
  then echo "Error - do not run this script as root."
  exit 1
fi
```
(from https://stackoverflow.com/a/18216122 , untested)

I think we can just exit. If the user really wants to execute as root for whatever bad reason, they just modify the script.

Reply:
I agree with you: actually though, it can run as root quite safely (but do not try it too much :-) , so I'm still not sure if running POH as root is an "error" or a "warning". @mwithi what do you think ?

Reply:
It may run fine, but strictly speaking root privileges are not needed, so for a well known security priciple (https://en.wikipedia.org/wiki/Principle_of_least_privilege), I'd just error and exit.